### PR TITLE
05 Basic Event Implementation

### PR DESCRIPTION
**Pull request by smazik02**
- Implemented a basic event entity with all essential fields
- Created an endpoint with basic CRUD operations

### CODE REVIEW

File: src/resource/events/event.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@Injectable()
+export class EventService {
+    constructor(
+        @InjectRepository(EventEntity)
+        private eventRepository: Repository<EventEntity>,
+    ) {}
+
+    async getEventById(eventId: number): Promise<EventEntity> {
+        const event = await this.eventRepository.findOneBy({ id: eventId });
+        if (!event) throw new NotFoundException('Note not found');
```
Comment (Aeri4a) [inline src/resource/events/event.service.ts:20]: Note not found :D

File: src/resource/events/event.service.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    ): Promise<any> {
+        const { affected } = await this.eventRepository.update(
+            {
+                id: noteId,
+            },
+            {
+                ...editEventDto,
+            },
+        );
+        if (affected == 0) throw new NotFoundException("Note doesn't exist");
```
Comment (Aeri4a) [inline src/resource/events/event.service.ts:51]: Michał will be angry

**Review by Aeri4a — CHANGES_REQUESTED** [review]
GJ. Small things, good start

File: src/resource/events/event.service.ts (hunk 2 above)
Comment (m-olej) [reply src/resource/events/event.service.ts:51]: >:(

File: src/app.module.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { TypeOrmModule } from '@nestjs/typeorm';
 import { UserModule } from './resource/user/user.module';
 import { AuthModule } from './auth/auth.module';
 import { NoteModule } from './resource/notes/notes.module';
+import { EventModule } from './resource/events/event.module';
```
[the remark this answers is not in the document]
Comment (smazik02) [reply src/app.module.ts:10]: I followed Michał's naming convention, I think we need a refactor for both in that case

File: src/database/entities/event.entity.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        name: 'created_at',
+        type: 'timestamp',
+    })
+    createdAt: Date;
+
+    @Column({ nullable: false })
+    createdById: number;
+
+    @ManyToOne(() => UserEntity, (user) => user.events)
+    createdBy: UserEntity;
```
[the remark this answers is not in the document]
Comment (smazik02) [reply src/database/entities/event.entity.ts:40]: This makes things 1000 times easier and I don't know of a more "elegant" solution

File: src/database/entities/user.entity.ts [hunk 5]
```diff
@@ -28,4 +29,7 @@ export class UserEntity {
 
     @OneToMany(() => NoteEntity, (note) => note.user)
     notes: NoteEntity[];
+
+    @OneToMany(() => EventEntity, (event) => event.createdBy)
+    events: EventEntity[];
```
[the remark this answers is not in the document]
Comment (smazik02) [reply src/database/entities/user.entity.ts:34]: It's more of a formality, I don't think we'll ever use it. I need to check if removing it breaks stuff

File: src/resource/events/dto/ReturnEvent.dto.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export class ReturnEventDto {
+    id: number;
+    name: string;
+    description?: string;
+    startDate: Date;
+    endDate?: Date;
```
[the remark this answers is not in the document]
Comment (smazik02) [reply src/resource/events/dto/ReturnEvent.dto.ts:6]: Maybe, I'll think about it

File: src/resource/events/event.service.ts (hunk 1 above)
Comment (smazik02) [reply src/resource/events/event.service.ts]: Fixed

File: src/resource/events/event.service.ts (hunk 2 above)
Comment (smazik02) [reply src/resource/events/event.service.ts]: :)

File: src/resource/events/dto/ReturnEvent.dto.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export class ReturnEventDto {
```
[the remark this answers is not in the document]
Comment (smazik02) [reply src/resource/events/dto/ReturnEvent.dto.ts]: No

File: src/database/entities/event.entity.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (smazik02) [reply src/database/entities/event.entity.ts]: I deleted them, they're not necessary

File: src/resource/events/dto/ReturnEvent.dto.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (smazik02) [reply src/resource/events/dto/ReturnEvent.dto.ts]: Done

File: src/app.module.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (smazik02) [reply src/app.module.ts]: Done

File: src/database/entities/user.entity.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (smazik02) [reply src/database/entities/user.entity.ts]: Deleted them both

File: src/resource/events/dto/CreateEvent.dto.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import {
+    IsBoolean,
+    IsDateString,
+    IsNotEmpty,
+    IsOptional,
+    IsString,
+} from 'class-validator';
+
+export class CreateEventDto {
```
[the remark this answers is not in the document]
Comment (smazik02) [reply src/resource/events/dto/CreateEvent.dto.ts]: Done

File: src/resource/events/dto/EditEvent.dto.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { IsBoolean, IsDateString, IsOptional, IsString } from 'class-validator';
+
+export class EditEventDto {
```
[the remark this answers is not in the document]
Comment (smazik02) [reply src/resource/events/dto/EditEvent.dto.ts]: Done

**Pull request merged**